lib/google_crawler_web/controllers/session_controller.ex

```diff
@@ -0,0 +1,39 @@
+defmodule GoogleCrawlerWeb.SessionController do
+  use GoogleCrawlerWeb, :controller
+
+  alias GoogleCrawler.Accounts
+  alias GoogleCrawler.Accounts.User
+
+  plug :check_user_session when action in [:new]
```

Comment:
The same plug is required for the registration page to prevent authenticated users to see the auth pages. 

I opted to have that concern in the Router and a separated plug